### ISSUE TITLE
Intersection of two overlapping polygons results in a single point

### ISSUE THREAD

**Issue by GetPastTheMonkey** [Toblerity/Shapely#1061]
## Expected behavior and actual behavior.

I have the following two polygons that I would like to intersect:

![shapely_intersection_wtf](https://user-images.githubusercontent.com/18332289/104132606-c588e580-537e-11eb-8019-37ad6823a063.png)

Expected behavior: The intersection of these two polygons should be a polygon (the area which is shaded red and yellow)

Actual behavior: The intersection results in a single point (the black dot in the image) 

## Steps to reproduce the problem.

The polygons are generated using data provided by OpenStreetMap. I am not able to provide you the script that generated the polygons. But I can give you the polygons:

Red polygon:
```
POLYGON ((8.944060548642172 47.18288466609015, 8.944430382046539 47.18148132698487, 8.944463831229625 47.18134565668537, 8.944560473190766 47.18092428414187, 8.944597208850295 47.18075030526124, 8.944642504479914 47.18051509992147, 8.944547560753543 47.1753515727357, 8.944530906765189 47.17515224660963, 8.944501284465577 47.17495404039642, 8.944472690280554 47.17482172198111, 8.94446914521204 47.17475605154142, 8.944165319022463 47.17218858066663, 8.94444145013478 47.17158009729027, 8.944491890699149 47.17146065293717, 8.94452193292447 47.17136099732583, 8.944784885369677 47.17038813527133, 8.944830386395591 47.17000193958746, 8.944854686586616 47.16974456549836, 8.945134260373642 47.16585967016191, 8.945129510784293 47.16573384047337, 8.944867576415186 47.16057644939821, 8.944329376345074 47.15134231861931, 8.943955553274689 47.11915097686187, 8.943671999999999 47.1193253, 8.9437347 47.119539, 8.943659 47.1197463, 8.9433445 47.1200405, 8.943236000000001 47.120244, 8.9432025 47.1206372, 8.9433568 47.121283, 8.942921 47.1218996, 8.942843099999999 47.1225536, 8.9429491 47.1228257, 8.9428576 47.1231874, 8.942712800000001 47.1234116, 8.9427854 47.1238173, 8.942312599999999 47.1244921, 8.9422906 47.1246334, 8.9423847 47.1250073, 8.9425752 47.1252287, 8.9425866 47.1253433, 8.9422432 47.1257778, 8.942263199999999 47.1259558, 8.942176099999999 47.1262903, 8.942236100000001 47.1264477, 8.9422525 47.1269369, 8.942612199999999 47.1272435, 8.9428511 47.1273161, 8.942868900000001 47.1274223, 8.9427889 47.1275655, 8.942494200000001 47.1277561, 8.9421567 47.1278314, 8.941975899999999 47.1279336, 8.941727 47.1282805, 8.941917999999999 47.1287181, 8.9417814 47.1289014, 8.941746200000001 47.1291028, 8.941330199999999 47.1292542, 8.941055499999999 47.1292938, 8.9407575 47.1294588, 8.9406546 47.1296, 8.9406391 47.1298723, 8.9405687 47.1299555, 8.9403749 47.1300533, 8.940104 47.1301094, 8.9389202 47.1301656, 8.9375734 47.1308215, 8.9372858 47.1310326, 8.937235299999999 47.1311691, 8.9369888 47.1313832, 8.935934700000001 47.1315943, 8.93519 47.1316651, 8.9349037 47.1317622, 8.934664 47.1320147, 8.934309799999999 47.1320198, 8.9340615 47.1320892, 8.9334238 47.1320806, 8.932900800000001 47.1321497, 8.9322602 47.1324191, 8.9321644 47.132618, 8.9320281 47.1327436, 8.9320532 47.1328399, 8.9319659 47.1329454, 8.931733599999999 47.133041, 8.9312206 47.1331316, 8.931057600000001 47.1332028, 8.930304400000001 47.1332474, 8.930190400000001 47.1332985, 8.9301961 47.1335525, 8.929846299999999 47.1344489, 8.9295288 47.1345724, 8.929281899999999 47.1348442, 8.928979200000001 47.1348957, 8.928592500000001 47.1350411, 8.9284038 47.1352433, 8.928484600000001 47.1354827, 8.9287806 47.1357091, 8.928796200000001 47.1358329, 8.928519100000001 47.1361197, 8.928247900000001 47.1362891, 8.927759200000001 47.1361208, 8.927576200000001 47.1361013, 8.9274793 47.1361996, 8.9273256 47.1362484, 8.926897 47.1361475, 8.9267065 47.1358888, 8.9263808 47.1357689, 8.9263212 47.1359524, 8.926155 47.1360726, 8.9259021 47.1361455, 8.925298 47.1360651, 8.924972800000001 47.1360766, 8.9247648 47.1361897, 8.924590200000001 47.136361, 8.9245094 47.1367144, 8.9241674 47.1368703, 8.9239181 47.1372249, 8.923778199999999 47.1373003, 8.923593 47.1378716, 8.923182799999999 47.1381848, 8.9227776 47.1382349, 8.922484300000001 47.1383862, 8.922077399999999 47.1384036, 8.9218882 47.138472, 8.9213059 47.1383825, 8.9209035 47.138404, 8.920705099999999 47.1386016, 8.920636999999999 47.1387775, 8.9206004 47.1388721, 8.9202771 47.1392622, 8.9201126 47.1392955, 8.919602299999999 47.1392752, 8.9194376 47.139194, 8.9184103 47.1390455, 8.9180718 47.1390895, 8.917791299999999 47.1391939, 8.917612699999999 47.1393753, 8.917650699999999 47.1394594, 8.917267799999999 47.1397489, 8.916593000000001 47.1401337, 8.9162865 47.1403785, 8.9160567 47.1404302, 8.915982945256452 47.14041649859894, 8.916026068677823 47.14167127612824, 8.916763840511631 47.14855934362856, 8.918208403753837 47.16154209122054, 8.917908352511699 47.16407449489247, 8.917283749185927 47.16741942856774, 8.917663060433433 47.165388105786, 8.920531859501343 47.17453483730279, 8.921026184744976 47.1751945715521, 8.921089898534561 47.17526234347078, 8.921090394051296 47.17526737241555, 8.92117692035041 47.17572161116221, 8.921154703034828 47.17956083382525, 8.921591673127663 47.18725475277926, 8.918927699999999 47.1880821, 8.923005699999999 47.1868156, 8.9282542 47.1848045, 8.930020900000001 47.1842038, 8.9324712 47.1835519, 8.933672518828848 47.18333686839551, 8.933555105660151 47.18159775225988, 8.933534120609211 47.18108703309196, 8.933243689261886 47.17879649032513, 8.933164032036636 47.17834635752828, 8.933135053340724 47.17812843047606, 8.932397133296767 47.17429517859337, 8.932470411377761 47.17349622047021, 8.932506779309117 47.17213274408402, 8.932498985568433 47.17192669260641, 8.931298250086611 47.17272705872695, 8.930816791852383 47.1729146187639, 8.930644164102819 47.17293101581292, 8.930150935248271 47.1732480714946, 8.929815957089398 47.17321836924309, 8.92979045174939 47.17324882024442, 8.929672938145352 47.1735938551449, 8.929670500939574 47.17359601775573, 8.929341665355697 47.17357515223639, 8.92917980371127 47.17374227954226, 8.929059569741232 47.17398578082324, 8.929082846732719 47.17413608386426, 8.928577384123171 47.17577127790207, 8.928503448926772 47.17557065323631, 8.928424273422667 47.17530774184235, 8.928324975550952 47.17485090664326, 8.928238311476042 47.17405300650063, 8.928245072460449 47.17400000783332, 8.92848084105133 47.17371622062752, 8.928518878666127 47.17338330329019, 8.928497675415102 47.17333446664745, 8.928972332030209 47.17311574007271, 8.928944996854803 47.1727070317643, 8.928964270721858 47.1726454725043, 8.929007712313933 47.17259512898877, 8.928971968757367 47.17237844372145, 8.92901997839129 47.17229605229427, 8.929138107914703 47.17132976913663, 8.929158200152367 47.17128476248578, 8.931988685339897 47.16583282349161, 8.933988146498786 47.16152803679944, 8.93399303194815 47.16300558429489, 8.934017176782284 47.16401551042161, 8.935841447081442 47.17084472662657, 8.935850686748772 47.17086976667597, 8.936299051744315 47.1717834311432, 8.936737599609145 47.17232951633346, 8.937143831388841 47.17298290353497, 8.937722429052847 47.17412658065862, 8.937977795635701 47.17448362896514, 8.937981332818291 47.17453892723807, 8.938203421335285 47.17565049498504, 8.937226641488019 47.18291910475116, 8.9381308 47.1828556, 8.940654200000001 47.1828033, 8.943463299999999 47.1828715, 8.943482599999999 47.1827335, 8.943625300000001 47.1827407, 8.943612 47.1828755, 8.944060548642172 47.18288466609015), (8.9282542 47.1848045, 8.924673447040355 47.1861765591173, 8.92462504733737 47.18516868977845, 8.924029508662979 47.17648555197359, 8.923723669331416 47.17545710067721, 8.923789114627194 47.17523527309795, 8.923501109679824 47.17500226612842, 8.923500378271475 47.17500216131309, 8.923467622881121 47.17495712274838, 8.923540909139879 47.17455967221756, 8.923665050192167 47.17447891630255, 8.923811485057078 47.17299696489082, 8.923847093250673 47.17302501070222, 8.924120279354604 47.17333352573859, 8.924253026716771 47.17358919664616, 8.924441306110991 47.17384323988766, 8.924680672519438 47.17388747467019, 8.925244985074546 47.17411969193819, 8.925315978018562 47.17404229443764, 8.925596356985938 47.17415789513893, 8.925646855180227 47.17430426896609, 8.925865975936745 47.17434811680337, 8.926006376550978 47.17479180087685, 8.92625528826456 47.17483868380299, 8.926965391079579 47.17677691939598, 8.928655081883193 47.18063640356721, 8.929034316339717 47.18453925073002, 8.9282542 47.1848045))
```

Yellow polygon:
```
POLYGON ((8.921089898534561 47.17526234347078, 8.921090394051296 47.17526737241555, 8.92117692035041 47.17572161116221, 8.921154703034828 47.17956083382525, 8.922365260364639 47.20087563348197, 8.923514023488929 47.20087563348197, 8.924117851561775 47.20087563348197, 8.924887591573489 47.20087563348197, 8.925379323095882 47.20087563348197, 8.92462504733737 47.18516868977845, 8.924029508662979 47.17648555197359, 8.923723669331416 47.17545710067721, 8.923789114627194 47.17523527309795, 8.923501109679824 47.17500226612842, 8.923500378271475 47.17500216131309, 8.923467622881121 47.17495712274838, 8.923540909139879 47.17455967221756, 8.923665050192167 47.17447891630255, 8.923811485057078 47.17299696489082, 8.923500244919989 47.1726657437983, 8.923369706739658 47.17249362466571, 8.923105734352891 47.17208651567921, 8.92301373702268 47.17192621713922, 8.922237343333293 47.16991046272432, 8.921630365927228 47.16804908562651, 8.92091189878685 47.16662138167527, 8.912715995830331 47.14961513924158, 8.920531859501343 47.17453483730279, 8.921026184744976 47.1751945715521, 8.921089898534561 47.17526234347078))
```

After loading the polygons, run `yellow_polygon.intersection(red_polygon)` which results in the intersection point `'POINT (8.9176630604334 47.165388105786)'` instead of a polygon.

## Operating system
Ubuntu 20.04 LTS
Python 3.8.5

## Shapely version and provenance
Shapely 1.8.dev0
Installed from GitHub using pip
(`pip install git+https://github.com/Toblerity/Shapely.git#egg=Shapely`)

**Comment by Jeremiah-England** [comment]
I can reproduce this with the exception that the red polygon appears to have a whole where yours does not. But I think that is mostly a distraction to the issue here since I get the same point you do. 

![image](https://user-images.githubusercontent.com/34973839/104133737-5c3fad00-5353-11eb-9371-94641118033d.png)

My specs are only slightly different from yours so I'll just include a GEOS version: `3.8.0-CAPI-1.13.1`.

Both polygons report as `is_valid` to me even though the red one clearly appears to be invalid.

**Comment by GetPastTheMonkey** [comment]
It seems I did not check for interiors, thanks for pointing out the hole in the polygon. But I agree, this should nonetheless return a polygon instead of a point.

My GEOS version is the same as yours: `3.8.0-CAPI-1.13.1`

**Comment by jorisvandenbossche** [comment]
I can also reproduce the single point result with latest Shapely (using GEOS 3.8). 
But testing in my environment with GEOS master, I now get a polygon. There have been quite some improvements in GEOS 3.9 to the overlay operations (including intersection), and so my guess is that this will be fixed in GEOS 3.9 (just released, but not yet used in the Shapely wheels or not yet available on conda).

The polgyon that I get is:

```
POLYGON ((8.918208 47.161542, 8.917908 47.164074, 8.917663 47.165388, 8.920532 47.174535, 8.921026 47.175195, 8.92109 47.175262, 8.92109 47.175267, 8.921177 47.175722, 8.921155 47.179561, 8.921592 47.187255, 8.923006 47.186816, 8.924673 47.186177, 8.924625 47.185169, 8.92403 47.176486, 8.923724 47.175457, 8.923789 47.175235, 8.923501 47.175002, 8.9235 47.175002, 8.923468 47.174957, 8.923541 47.17456, 8.923665 47.174479, 8.923811 47.172997, 8.9235 47.172666, 8.92337 47.172494, 8.923106 47.172087, 8.923014 47.171926, 8.922237 47.16991, 8.92163 47.168049, 8.920912 47.166621, 8.918132 47.160853, 8.918208 47.161542))
```

**Comment by jorisvandenbossche** [comment]
Plot of the result (resulting polygon is hatched area):

![Figure_1](https://user-images.githubusercontent.com/1020496/104601998-34b25280-567b-11eb-9b4d-0242248c4a25.png)

**Comment by dr-jts** [comment]
I confirm that this was an issue in GEOS versions <= 3.8, and is is fixed in GEOS 3.9.

**Comment by jorisvandenbossche** [comment]
I opened a PR to add this as a test case to GEOS: https://github.com/libgeos/geos/pull/385

This will be fixed once new Shapely packages are released using GEOS 3.9, for the rest nothing to do about it in Shapely itself, so closing this issue. 

@GetPastTheMonkey thanks for the report!
